lib/graph3d/Settings.js

```diff
@@ -467,6 +469,50 @@ function setDataColor(dataColor, dst) {
   }
 }
 
+/**
+ * Converts an object that accepts a HUE keyword. This is converted to a certain amount of hex color stops. At which point:
+ * the HTML hex color codes is converted into an RGB color object.
+ * 
+ * @param {Object | Array<string>} surfaceColors Either an object that describes the HUE, or an array of HTML hex color codes
+ * @param {Object} dst 
+ */
+function setSurfaceColor(surfaceColors, dst) {
+  if(surfaceColors === undefined) {
+    return;    // Nothing to do
+  }
+
+  if (dst.surfaceColors === undefined) {
+    dst.surfaceColors = {};
+  }
+
+  let rgbColors = [];
+
+  if(Array.isArray(surfaceColors)) {
+    rgbColors = surfaceColors.map(util.hexToRGB);
+  } else if (typeof surfaceColors === 'object') {
+    if(surfaceColors.hue === undefined) {
+      throw new Error('Unsupported type of surfaceColors');
+    }
+
+    const startHue = surfaceColors.hue.start;
+    const endHue = surfaceColors.hue.end;
+    const saturation = surfaceColors.hue.saturation;
+    const brightness = surfaceColors.hue.brightness;
+    const stops = surfaceColors.hue.colorStops;
+    const hueStepSize = (endHue - startHue) / stops;
+
+    let currentHue = startHue;
+    while (currentHue < endHue) {
+      rgbColors.push(util.HSVToRGB(currentHue/360, saturation/100, brightness/100));
```

Comment:
Changing this line to:
```javascript
rgbColors.push(
  util.HSVToRGB(
    (360 + (currentHue % 360)) / 360,
    saturation / 100,
    brightness / 100
  )
);
```
will allow for the hue to wrap around. Giving users the option to go from cyan through red to green or even to repeat hues. At the same time this solves the issue when the colors glitch for negative hue and anything above 360 is silently ignored.